xmtp_content_types/src/compatibility_test.rs

```diff
@@ -0,0 +1,187 @@
+                        obj.insert("networkId".to_string(), Value::String(n.to_string()));
+                    }
+                }
+            }
+        }
+
+        assert_eq!(
+            original_json,
+            re_encoded_json,
+            "JSON content mismatch for {} ({}): {}",
+            content_type,
+            index + 1,
+            description
+        );
+    }
+
+    /// Decodes and re-encodes content using the appropriate codec
+    fn decode_and_reencode(
+        content_type: &str,
+        encoded_content: &EncodedContent,
+    ) -> Result<EncodedContent, Box<dyn std::error::Error>> {
+        match content_type {
+            "text" => {
+                use crate::text::TextCodec;
+                let content = TextCodec::decode(encoded_content.clone())?;
+                Ok(TextCodec::encode(content)?)
+            }
+            "reaction" => {
+                use crate::reaction::LegacyReactionCodec;
+                let content = LegacyReactionCodec::decode(encoded_content.clone())?;
+                Ok(LegacyReactionCodec::encode(content)?)
+            }
+            "reply" => {
+                use crate::reply::ReplyCodec;
+                let content = ReplyCodec::decode(encoded_content.clone())?;
+                Ok(ReplyCodec::encode(content)?)
+            }
+            "readReceipt" => {
+                use crate::read_receipt::ReadReceiptCodec;
+                let content = ReadReceiptCodec::decode(encoded_content.clone())?;
+                Ok(ReadReceiptCodec::encode(content)?)
+            }
+            "remoteAttachment" => {
+                use crate::remote_attachment::RemoteAttachmentCodec;
+                let content = RemoteAttachmentCodec::decode(encoded_content.clone())?;
+                Ok(RemoteAttachmentCodec::encode(content)?)
+            }
+            "transactionReference" => {
+                use crate::transaction_reference::TransactionReferenceCodec;
+                let content = TransactionReferenceCodec::decode(encoded_content.clone())?;
+                Ok(TransactionReferenceCodec::encode(content)?)
+            }
+            "groupUpdated" => {
+                use crate::group_updated::GroupUpdatedCodec;
+                let content = GroupUpdatedCodec::decode(encoded_content.clone())?;
+                Ok(GroupUpdatedCodec::encode(content)?)
+            }
+            "attachment" => {
+                use crate::attachment::AttachmentCodec;
+                let content = AttachmentCodec::decode(encoded_content.clone())?;
+                Ok(AttachmentCodec::encode(content)?)
+            }
+            _ => Err(format!("Unsupported content type: {content_type}").into()),
+        }
+    }
+
+    #[test]
+    fn integration_test() {
+        let fixtures_path = Path::new("fixtures/serialized_content.json");
+        let json_content =
+            fs::read_to_string(fixtures_path).expect("Failed to read serialized_content.json");
```

Comment:
Will put up a PR in xmtp-js for regenerating this fixture based on the JS content types